src/agora/dofft.cc

```diff
@@ -192,23 +195,24 @@ EventData DoFFT::Launch(size_t tag) {
   } else if (sym_type == SymbolType::kUL) {
     PartialTranspose(cfg_->GetDataBuf(data_buffer_, frame_id, symbol_id),
                      ant_id, SymbolType::kUL);
-  } else if (sym_type == SymbolType::kCalUL &&
-             ant_id != cfg_->RefAnt(cell_id)) {
+  } else if (sym_type == SymbolType::kCalUL) {
     // Only process uplink for antennas that also do downlink in this frame
     // for consistency with calib downlink processing.
     if (frame_id >= TX_FRAME_DELTA &&
         ant_id / cfg_->AntPerGroup() ==
             (frame_id - TX_FRAME_DELTA) % cfg_->AntGroupNum()) {
-      size_t frame_grp_id = (frame_id - TX_FRAME_DELTA) / cfg_->AntGroupNum();
-      size_t frame_grp_slot = frame_grp_id % kFrameWnd;
+      const size_t frame_grp_id =
+          (frame_id - TX_FRAME_DELTA) / cfg_->AntGroupNum();
+      const size_t frame_grp_slot = frame_grp_id % kFrameWnd;
       PartialTranspose(
           &calib_ul_buffer_[frame_grp_slot][ant_id * cfg_->OfdmDataNum()],
           ant_id, sym_type);
       phy_stats_->UpdateCalibPilotSnr(frame_grp_id, 1, ant_id, fft_inout_);
     }
-  } else if (sym_type == SymbolType::kCalDL &&
-             ant_id == cfg_->RefAnt(cell_id)) {
-    if (frame_id >= TX_FRAME_DELTA) {
+    RtAssert(radio_id != cfg_->RefRadio(cell_id),
+             "Received a Cal Ul symbol for an antenna on the reference radio");
+  } else if (sym_type == SymbolType::kCalDL) {
+    if ((ant_id == cfg_->RefAnt(cell_id)) && (frame_id >= TX_FRAME_DELTA)) {
       size_t frame_grp_id = (frame_id - TX_FRAME_DELTA) / cfg_->AntGroupNum();
```

Comment:
As mentioned above, perhaps we can also rename AntGroupNum to something like FramePerCalibRound or FramesPerFullCalib or similar, for better readability?

Reply:
Will fix in next PR